pkg/storaged/lvol-tabs.jsx

```diff
@@ -74,12 +77,16 @@ function lvol_and_fsys_resize(client, lvol, size, offline, passphrase) {
             } else {
                 return fsys.Resize(size - crypto_overhead, { });
             }
+        } else if (vdo) {
+            if (size - crypto_overhead > vdo.physical_size)
+                return vdo.grow_physical();
+            else if (size - crypto_overhead < vdo.physical_size)
+                return cockpit.reject("Trying to shrink VDO.  Ignored.");
+            else
+                return cockpit.resolve();
         } else if (size < orig_size) {
             // This shouldn't happen.  But if it does, continuing is harmful, so we throw an error.
-            console.warn("Trying to shrink unrecognized content.  Ignored.");
-            return cockpit.reject();
-        } else if (vdo) {
-            return vdo.grow_physical();
+            return cockpit.reject("Trying to shrink unrecognized content.  Ignored.");
```

Comment:
Hmm, this should not `reject`, just warn.

Reply:
Correct.  This should never happen since the setup code will make sure we don't end up here.  But there might be bugs, and also the setup rnus before the dialog opens, and the system might look very different when the action is actually carried out.

Reply:
Upps. I was confused, sorry.  We should `reject` attempts to shrink unrecognized content (as well as attempts to shrink VDO).  Attempting to shrink unshrinkable filesystems is okay, since we will get a error message from Filesystem.Resize.

This PR just moved the error message from the console into the dialog.  I will make this a separate commit and remove the word "Ignored" from the error.

In all these cases, the UI will prevent the user from initiating these actions (e.g. the "Shrink" button is disabled for unrecognized content), but if the storage configuration changes enough while the dialog is open, we might still end up in that code branch.

This is a general issue with all dialogs, and I have plans how to make the code much nicer for this...